plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -115,15 +117,16 @@ void setSiteUri(URI siteUri) {
   }
 
   public synchronized List<LibraryFile> getLibraryFiles() {
```

Comment:
Returning different results whether a library is resolved or not seems a bit dangerous, and counter to the model (the extension point has `<library><libraryFile/><libraryFile/>…</library>`.  If you changed `getDirectDependencies()` → `getTransitiveDependencies()` you could get rid of the resolved flag and use whether `transitiveDependencies == null` as the flag.

Reply:
I do think this class is do for some more rework since it was desigbned under the assumption (no longer valid) that most libraries listed all their deps in plugin.xml. resolved should probably default to false. 

That said, I'm not sure I want to change more than the minimum we need until 1.4.0 is  out. It feels like it risks introducing other bugs.

Reply:
That's my point: changing `getLibraryFiles()` to return the transitive dependencies is counter to the original design, and it has introduced bugs.

Reply:
getLibraryFiles() originally returned the complete graph of dependencies. I don't think that's changed. 

Let me explore using `transitiveDependencies == null` as the flag. For sanity, I'll probably rename the two methods to 

```
getDirectDependencies
getAllDependencies
```